source/Ship.h

```diff
@@ -365,7 +369,8 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// DamageDealt from that weapon. The return value is a ShipEvent type,
 	// which may be a combination of PROVOKED, DISABLED, and DESTROYED.
 	// Create any target effects as sparks.
-	int TakeDamage(std::vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment);
+	int TakeDamage(std::vector<Visual> &visuals, const DamageDealt &damage, const Government *sourceGovernment,
+		const Point &damageSource = Point());
```

Comment:
So for hazards it is system center?
If yes, that would probably contradict hazards with "constant strength"
But that's only a minor detail